tests/run-make/rust-lld-compress-debug-sections/rmake.rs

```diff
@@ -0,0 +1,39 @@
+// Checks the `compress-debug-sections` option on rust-lld.
+
+//@ needs-rust-lld
+//@ only-linux
+//@ ignore-cross-compile
+
+// FIXME: This test isn't comprehensive and isn't covering all possible combinations.
+
+use run_make_support::{assert_contains, cmd, llvm_readobj, run_in_tmpdir, rustc};
+
+fn check_compression(compression: &str, to_find: &str) {
+    run_in_tmpdir(|| {
+        let out = rustc()
+            .arg("-Zlinker-features=+lld")
+            .arg("-Clink-self-contained=+linker")
+            .arg("-Zunstable-options")
+            .arg("-Cdebuginfo=full")
+            .link_arg(&format!("-Wl,--compress-debug-sections={compression}"))
+            .input("main.rs")
+            .run_unchecked();
+        let stderr = out.stderr_utf8();
+        if stderr.is_empty() {
+            llvm_readobj().arg("-t").arg("main").run().assert_stdout_contains(to_find);
```

Comment:
Suggestion: since "llvm-readobj provides -t as an alias for [--symbols](https://llvm.org/docs/CommandGuide/llvm-readobj.html#cmdoption-llvm-readobj-symbols)", this can just use the `symbols()` helper.

Reply:
-t is an alias for --section-details, not --symbols.

Reply:
Are we sure? https://llvm.org/docs/CommandGuide/llvm-readobj.html#differences-to-llvm-readelf says

> llvm-readobj provides -t as an alias for [--symbols](https://llvm.org/docs/CommandGuide/llvm-readobj.html#cmdoption-llvm-readobj-symbols). llvm-readelf does not.

It's possible that the llvm-readobj docs are wrong?

Reply:
```
khuey@U-2ZRST2SK7QTBO:/pernosco/llvm-project$ llvm-readobj -t /usr/bin/true | md5sum
6e25191991ccaf02a636c1cb6d86598d  -
khuey@U-2ZRST2SK7QTBO:/pernosco/llvm-project$ llvm-readobj --section-details /usr/bin/true | md5sum
6e25191991ccaf02a636c1cb6d86598d  -
khuey@U-2ZRST2SK7QTBO:/pernosco/llvm-project$ llvm-readobj --symbols /usr/bin/true | md5sum
85a41add8f07c443c3fbacbc2dfe5104  -
```

Reply:
Thanks for confirming. I also checked this locally, apparently those docs are just outdated, https://reviews.llvm.org/D105055?id=354996 I think changed this behavior to match llvm-readelf in like 2021 lol.